### PR TITLE
Add `isTestnet` flag to `Chain` model

### PR DESCRIPTION
**Pull request by iamacook**
This adds a new `isTestnet` `boolean` flag to the `Chain` model and updates associated tests.

Note: the `test_on_feature_update_with_multiple_safe_apps` test began to fail when running all test suites. I've found a temporary solution and added a comment for us to revisit.

### CODE REVIEW

**Comment by iamacook** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/safe_apps/tests/test_signals.py [hunk 1]
```diff
@@ -522,6 +523,8 @@ def test_on_feature_delete_with_safe_app(self) -> None:
             "utf-8"
         )
 
+    # TODO: this will pass if moved to the top of the file suggesting some kind of mock leak
```
Comment (mmv08) [inline src/safe_apps/tests/test_signals.py:526]: this is scary. are you going to fix it in this pr?

**Review by mmv08 — APPROVED** [review]
British quality code 🇬🇧 🏴󠁧󠁢󠁥󠁮󠁧󠁿

File: src/safe_apps/tests/test_signals.py [hunk 2]
```diff
@@ -522,6 +522,8 @@ def test_on_feature_delete_with_safe_app(self) -> None:
             "utf-8"
         )
 
+    # Otherwise fails when testing with all suites - above tests somehow leak
+    @responses.stop  # type: ignore
```
Comment (fmrsabino) [inline src/safe_apps/tests/test_signals.py:526]: Discussed internally and we will tackle this separately 👍

**Review by fmrsabino — APPROVED** [review]
🚀

**Pull request merged**